Detectors/ITSMFT/ITS/simulation/src/DigitChip.cxx

```diff
@@ -0,0 +1,74 @@
+//
+//  DigitChip.cxx
+//  ALICEO2
+//
+#include "ITSSimulation/DigitChip.h"
+#include "ITSBase/Digit.h"
+
+#include "FairLogger.h"  // for LOG
+
+#include "TClonesArray.h"
+
+using namespace AliceO2::ITS;
+
+Int_t DigitChip::fnRows=650;
+
+DigitChip::DigitChip()
+{
+}
+
+DigitChip::~DigitChip()
+{
+  Reset();
+}
+
+void DigitChip::Reset()
+{
+  for (auto pixel: fPixels) {
+    delete pixel.second;
+  }
+  fPixels.clear();
+}
+
+Digit *DigitChip::FindDigit(Int_t idx)
+{
+  Digit *result = nullptr;
+  auto digitentry = fPixels.find(idx);
+  if (digitentry != fPixels.end()) {
+    result = digitentry->second;
+  }
+  return result;
+}
+
+Digit *DigitChip::GetDigit(UShort_t row, UShort_t col) {
+  Int_t idx = col*fnRows + row;
+  return FindDigit(idx); 
+}
+
+Digit *DigitChip::AddDigit(
+       UShort_t chipid, UShort_t row, UShort_t col,
+       Double_t charge, Double_t timestamp
+)
+{
+  Int_t idx = col*fnRows + row;
+
+  Digit *digit=FindDigit(idx);
+  if (digit) {
+    LOG(DEBUG) << "Adding charge to pixel..." << FairLogger::endl;
+    charge += digit->GetCharge();
+    delete digit;
+  }
+  
+  digit = new Digit(chipid,row,col,charge,timestamp);
+  fPixels.insert(std::pair<Int_t, Digit *>(idx,digit));
+
```

Comment:
@sawenzel, @iouribelikov 
I think we should avoid (even: ban) naked new's and deletes because of zero ownership information. In fact i think nowadays there is little (zero) need for even a "placement new" in algorithms. TClonesArray is something we are all used to, but in my opinion we should use the more modern stuff, e.g. a std::vector(with ::emplace_back()) or something like that. TClonesArray() preallocation model is handled by the new standard in a much better way. We could even use some other allocators with the std:: stuff, like a stack based allocator with memory handled by some object in an upper scope (by a proper owner: unique_ptr, vector, whatever, anything but ownig raw pointers :) ).
cheers,
Mikolaj